analyzers/src/SonarAnalyzer.CSharp/Rules/MemberShouldBeStatic.cs

```diff
@@ -184,20 +179,30 @@ private static bool IsLeftmostIdentifierName(ExpressionSyntax node)
             var conditional = node.Parent as ConditionalAccessExpressionSyntax;
             var memberBinding = node.Parent as MemberBindingExpressionSyntax;
 
-            return (memberAccess == null && conditional == null && memberBinding == null)
+            return (memberAccess is null && conditional is null && memberBinding is null)
                 || memberAccess?.Expression == node
                 || conditional?.Expression == node;
         }
 
-        private static bool IsInstanceMember(ExpressionSyntax node, SemanticModel semanticModel)
+        private static bool IsInstanceMember(ExpressionSyntax node, SemanticModel model)
         {
             if (node is InstanceExpressionSyntax)
             {
                 return true;
             }
-
-            return semanticModel.GetSymbolInfo(node).Symbol is { IsStatic: false } symbol
-                   && InstanceSymbolKinds.Contains(symbol.Kind);
+            // For ctor(foo: bar), 'IsConstructorParameter(foo)' returns true. This check prevents that case.
+            if (node.Parent is NameColonSyntax)
```

Comment:
this is a bit annoying but if you call a constructor with a named argument, you get this tree:

![image](https://github.com/SonarSource/sonar-dotnet/assets/115458417/15dac6c3-58a0-4dd2-967e-e76fb7c2e1ca)

`paramName` passes all the checks:
- it is a parameter symbol
- the method is a constructor

So it generates an FP.
The more semantically correct way to check this is to go up the tree from `node` and find the first `MemberDeclarationSyntax`, fetch the symbol, and compare if the `.ContainingType` is the same as the symbol itself.
...but that's more expensive than a simple syntax check, and this way we also prune some checks that are unnecessary (`NameColon` is always irrelevant for this rule)